### PR TITLE
Update ES7 proposal processes for individual tests; add pre-strawman

### PR DESCRIPTION
**Pull request by ljharb**
- Update ES7 proposal processes for individual tests
- Add "pre-strawman" for proposals that are not listed on https://github.com/tc39/ecma262/blob/master/stage0.md or https://github.com/tc39/ecma262/blob/master/README.md (per https://github.com/kangax/compat-table/pull/548#issuecomment-106106401)

Anything not yet on the `ecma262` repo shouldn't be considered "required", since it's nowhere close to being included in the spec yet.

### CODE REVIEW

**Comment by webbedspace** [comment]
One problem: "annex b" currently produces cells bearing the message "Optional on non-browser platforms" for only non-browser platforms, so extending it to mean "pre-stage-0" is a bit dubious, given that arguably pre-stage-0 should be optional for all platforms, including browsers.

**Comment by ljharb** [comment]
Ah, true. I'll improve that real quick :-)

**Comment by ljharb** [comment]
@webbedspace updated :-D

File: build.js [hunk 1]
```diff
@@ -31,6 +31,10 @@ var child_process = require('child_process');
 
 var useCompilers = String(process.argv[2]).toLowerCase() === "compilers";
 
+var isOptional = function isOptional(category) {
```
Comment (kangax) [inline build.js:34]: why not just `function isOptional() { ... }`? :)

Comment (ljharb) [inline build.js:34]: Just to avoid hoisting - if you have a strong preference I'll be happy to change it to a declaration.

Comment (kangax) [inline build.js:34]: I just always prefer function declarations unless in blocks. Not a big deal, was just curious. LGTM, merging.

**Comment by ljharb** [comment]
Thanks!

**Comment by zloirock** [comment]
Additional category for pre-strawman - fine, but with current background I don't see where `Yes`, where `No` =\

**Comment by webbedspace** [comment]
I pushed a change now.

**Comment by zloirock** [comment]
@webbedspace thanks :)

**Pull request merged**